tagger/api.py

```diff
@@ -78,20 +78,37 @@ def endpoint_interrogate(self, req: models.TaggerInterrogateRequest):
         if req.image is None:
             raise HTTPException(404, 'Image not found')
 
+        if isinstance(req.model, list):
+            if any(i not in utils.interrogators.keys() for i in req.model):
+                raise HTTPException(404, 'Model {i} not found')
+
         if req.model not in utils.interrogators.keys():
             raise HTTPException(404, 'Model not found')
+            req.model = [req.model]
 
         image = decode_base64_to_image(req.image)
-        interrogator = utils.interrogators[req.model]
-
-        with self.queue_lock:
-            QData.tags.clear()
-            QData.ratings.clear()
-            QData.in_db.clear()
-            QData.for_tags_file.clear()
-            data = ('', '', '') + interrogator.interrogate(image)
+        QData.tags.clear()
+        QData.ratings.clear()
+        QData.in_db.clear()
+        QData.for_tags_file.clear()
+
+        # allow overriding of default values
+        if req.threshold:
+            QData.threshold = req.threshold
+        if req.tag_frac_threshold:
+            QData.tag_frac_threshold = req.tag_frac_threshold
+        if req.count_threshold:
+            QData.count_threshold = req.count_threshold
```

Comment:
this limits the total number of tags, excluded_tags to the topmost count_threshold count. Set to extremely higher value than nr of tags available to disable.